support-figma/extended-layout-plugin/src/code.ts

```diff
@@ -949,27 +949,48 @@ else if (figma.command === "move-plugin-data") {
       enabled: msg.enabled,
       discrete: msg.discrete,
       discreteValue: msg.discreteValue,
+      vertical: msg.vertical,
     }
     
     let meterData: any = {};
     if (progressMarker) {
       // The progress marker means we don't resize the node; we just move it along the x axis
-      const startX = -node.width / 2;
-      const endX = parent.width - node.width / 2;
-      const moveX = percentToValue(value, startX, endX);
-      node.x = moveX;
+      // if horizontal or y axis if vertical
```

Comment:
Especially here, we should try to make this as generic as we can.

Also, is there a way that we can make it completely generic and not dependent on "height" and "width"? Could there be a way for the user to designate the start and end points within the node? I really don't know if that's feasible, but if there were something like that I think it'd be much more intuitive.

Reply:
Making this completely generic would be hard, especially taking into account the fact that you can transform these rectangles and frames in Figma. I think what we have is sufficient for most if not all progress bar scenarios currently. Sure there may be some complicated and obscure scenarios we don't support yet. If we really want to revamp how this is done we should get @aubreydagoog and/or Richard involved with a redesign. I was just trying to get this in quickly since it's a relatively simple feature that would have a lot of benefit since multiple users have had issues building vertical progress bars with our current plugin.

Reply:
I think the "totally generic version" would just be growing a path. We've previously discussed using the stroker to generate a single dash that grows/shrinks along a path (where only the dash is filled). This would create the illusion of the path growing/shrinking, but it requires some careful attention on things like supporting stroke caps (rounded, square, butt, etc).

Currently, I think we still need all of the various options, so I think this change is a good add. In the future, we might be able to make the path be a default, but we'd need to be able to support moving a marker along the path, and we'd need it to feel acceptable to the person operating Figma.

Reply:
Moving along a path would be great! We'll add it to the list!